generator/standalone.cc

```diff
@@ -84,45 +68,49 @@ void ParseDelimitedArg(const std::string& arg,
 //   this is especially useful for folder operations, since standard library
 //   below C++17 does not have folder I/O abstraction whatsoever.
 //
-bool ConvertCommandLineArgs(int argc, const char* const argv[],
+bool ConvertCommandLineArgs(int argc, char const* const argv[],
                             std::vector<std::string>& args,
                             std::string* error_msg) {
   // GAPIC Generator Standalone arguments
-  const std::string desc_arg("--descriptor");
-  const std::string output_arg("--output");
-  const std::string package_arg("--package");
+  std::string const desc_arg("--descriptor");
+  std::string const output_arg("--output");
+  std::string const package_arg("--package");
 
-  const std::string desc_set_in_arg("--descriptor_set_in=");
+  std::string const desc_set_in_arg("--descriptor_set_in=");
 
   std::vector<std::string> desc_set_in;
   std::vector<std::string> packages;
 
   for (int i = 0; i < argc; i++) {
-    std::string arg(argv[i]);
-    auto pos = arg.find('=', 0);
-    if (pos == std::string::npos) {
-      args.emplace_back(arg);
+    std::vector<std::string> arg =
+        absl::StrSplit(argv[i], absl::MaxSplits('=', 1));
+    if (arg.size() <= 1) {
+      args.emplace_back(argv[i]);
       continue;
     }
 
-    const std::string& arg_name = arg.substr(0, pos);
-    const std::string& arg_val = arg.substr(pos + 1, arg.size() - pos);
+    std::string const& arg_name = arg[0];
+    std::string const& arg_val = arg[1];
 
     if (arg_name == desc_arg || arg_name == desc_set_in_arg) {
       args.emplace_back(desc_set_in_arg + arg_val);
-      ParseDelimitedArg(arg_val, desc_set_in);
+      std::vector<std::string> const& spl =
```

Comment:
Just curious: why make spl a const ref instead of a value, here and below?

Reply:
Both would work and both would be long enough to not fit in 80 char limit =(. Assigning the return value to a const reference will allow extend life time of the returned value to the lifetime of the reference itself. This, theoretically, should make it easier for compilers to optimize and avoid unnecessary copying (although they may already be doing it in this case, https://isocpp.org/wiki/faq/ctors#return-by-value-optimization). Non-const referrence would cause compilation error.

Reply:
With C++17 copy-elision and the return value optimization become mandatory. It's really infuriating that they're optional in C++11/14 because it forces us to be conservative in situations like these.